### DISCUSSION TITLE
how to recieve file in post request inside the body?

### DISCUSSION THREAD

**Discussion by gokulgseven** [pocketbase/pocketbase#4279] [Q&A]
I want to receive a file in pocketbase that is sent as a raw binary in the post request. 

I have a application that sends the file in this way.

```
var options = {
  method: 'POST',
  headers: {"Content-Type":"application/pdf"},
  body: "<file contents here>",
  redirect: 'follow'
};

fetch("https://url/file/upload", options)
```

i am trying to make a custom endpoint in pocketbase (extend with JavaScript) but where is the file saved. 

```
routerAdd("POST",` "/file/upload'", async (c) => {
    try {
        let data = c.request()
        console.log(JSON.stringify(data), "context")
        console.log(JSON.stringify(data.body), "body")
    }catch(error){
        console.log(error)
        return c.html(200, `<div>${error}</div>`)

    }
})
```
this doesnt seem to be working , i cant seem to locate the file in the request

**Comment by ganigeorgiev** [comment]
Your question lack details to understand why you want a custom action and what actually you are trying to do with the file (where the file will be stored, how it will be retrieved, how you'll perform validations for it, why the [builtin file upload](https://pocketbase.io/docs/files-handling/#uploading-files) doesn't work for your use case, etc).

Sending the request as `content-type:application/pdf` also doesn't seem correct to me as I'm not sure what is the expected behavior here (we don't have special handling for this type of request content-type).

Files are usually uploaded as `multipart/form-data` request. Since you are using a plain `fetch` call in this case I expect the body to be a [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) value and the file value to be a [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) or [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/FormData).

From the route you can access the parsed request file using `c.formFile("something")`.

But I don't recommend this and it has a lot of caveats as mentioned in the beginning and, unless you have a really specific reason wanting to do it that way, a better approach would be to use the [builtin collection `file` field handling](https://pocketbase.io/docs/files-handling/).

**Comment by gokulgseven** [comment]
I want to upload the file and save it in the built in collection only. 

the only reason I am trying to make a custom Api for it is because (I want to integrate it with a tally ERP which is an old application and it only supports file upload as raw binary in post body, (it doesn't support multiparts/form-data ))

(basically what I am looking for a middleware that can take the file that is supplied in post body and convert it to multipart/form-data )

Its a one-off use case. 
I was hoping that there is a function or method already, that will allow me to access the file contents from the post body and save that in a collection? 

(otherwise, i will have to make a separate server, that just takes it the request from tally ERP and converts file from post body to multipart and then sends it to pocket base )
